### PR TITLE
This changes the attributes to use the kamon environment, rather than trying to pull…

### LINKED ISSUE #31: Handle host="auto" correctly
It looks like we might need to do the work to look up localhost in the "auto" case, unless that's exposed by the kamon config/reporter API somewhere we haven't found yet.

### PR DESCRIPTION
**Pull request by jkwatson**
… it out of the config.

There are also a few bits in here to make a bit more compatible with older scala versions.

resolves #31

### CODE REVIEW

File: src/main/scala/kamon/newrelic/AttributeBuddy.scala [hunk 1]
```diff
@@ -38,18 +42,12 @@ object AttributeBuddy {
...
       .put("instrumentation.source", "kamon-agent")
-      .put("service.name", serviceName)
-      .put("host", host)
-    if (environment.hasPath("tags")) {
-      val environmentTags = environment.getConfig("tags")
-      AttributeBuddy.addTagsFromConfig(environmentTags, attributes)
-    }
+      .put("service.name", environment.service)
+      .put("host", environment.host)
+    AttributeBuddy.addTagsFromTagSets(Seq(environment.tags), attributes)
```
Comment (breedx-nr) [inline src/main/scala/kamon/newrelic/AttributeBuddy.scala]: definitely a clear improvement.

File: src/main/scala/kamon/newrelic/metrics/NewRelicMetricsReporter.scala [hunk 2]
```diff
@@ -62,13 +70,11 @@ object NewRelicMetricsReporter {
       new NewRelicMetricsReporter()
   }
 
-  def buildSender(): MetricBatchSender = {
-    val config = Kamon.config();
+  def buildSender(config: Config = Kamon.config()): MetricBatchSender = {
```
Comment (breedx-nr) [inline src/main/scala/kamon/newrelic/metrics/NewRelicMetricsReporter.scala]: 👍

File: src/main/scala/kamon/newrelic/metrics/NewRelicMetricsReporter.scala [hunk 3]
```diff
@@ -62,13 +70,11 @@ object NewRelicMetricsReporter {
...
     val nrConfig = config.getConfig("kamon.newrelic")
     val nrInsightsInsertKey = nrConfig.getString("nr-insights-insert-key")
     SimpleMetricBatchSender.builder(nrInsightsInsertKey)
       .enableAuditLogging()
-      .secondaryUserAgent("newrelic-kamon-reporter", LibraryVersion.version)
```
Comment (jkwatson) [inline src/main/scala/kamon/newrelic/metrics/NewRelicMetricsReporter.scala]: oh crud. this shouldn't go away. will fix.

Comment (jkwatson) [reply src/main/scala/kamon/newrelic/metrics/NewRelicMetricsReporter.scala]: done

Comment (breedx-nr) [reply src/main/scala/kamon/newrelic/metrics/NewRelicMetricsReporter.scala]: sorry I didn't catch that!

**Pull request merged**